expcore/auto_play.lua

```diff
@@ -0,0 +1,519 @@
+            if choice ~= 0 then
+                votes[choice] = votes[choice] + 1
+            end
+        end
+    end
+
+    return votes
+end
+
+--- Counts up and returns the value selected by vote for a given category
+local function select_by_votes(category)
+    local votes = count_votes(category)
+    local max_vote = math.max(unpack(votes))
+
+    -- Find all entries with max_votes votes.
+    local tied = {}
+    for i, cast_votes in pairs(votes) do
+        if cast_votes == max_vote then
+            tied[#tied + 1] = i
+        end
+    end
+
+    -- Pick a random from the tied votes if multiple
+    local index = #tied == 1 and tied[1] or tied[math.random(#tied)]
+
+    -- If the category has a index to value map return the mapped value.
+    if vote_categories[category].values then
+        return vote_categories[category].values[index]
+    end
+
+    -- Otherwise return the index that won the vote.
+    return index
+end
+
+--- Create list-box list for VoteSelector
+local function vote_selector_list(category)
+    local votes = count_votes(category)
+
+    -- Create list-box items with the counted votes
+    local list_items = {}
+    for i, item in ipairs(vote_categories[category].items) do
+        list_items[#list_items + 1] = {"", votes[i], " - ", item}
+    end
+    return list_items
+end
+
+-- Update vote counts for all categories and selectors
+local function recount_votes()
+    for category in pairs(vote_categories) do
+        local list_items = vote_selector_list(category)
+        for _, player in pairs(game.connected_players) do
+            local selector = player_data[player.index].vote_selectors[category]
+            if selector and selector.valid then
+                selector.items = list_items
+            end
+        end
+    end
+end
+
+--- Vote selector for a category.  Updates vote counts in real time.
+local VoteSelector =
+Gui.element(function(event_trigger, parent, data, category)
+    local category_data = vote_categories[category]
+
+    -- Wrapper flow to allow multiple selectors in the same flow.
+    local flow = parent.add{
+        type = 'flow',
+        direction = 'vertical',
+    }
+    flow.style.padding = 0
+
+    flow.add{
+        type = 'label',
+        caption = category_data.title,
+        style = 'heading_3_label',
+    }
+
+    local selector = flow.add{
+        name = event_trigger,
+        type = 'list-box',
+        items = vote_selector_list(category),
+        style = 'list_box_in_shallow_frame'
+    }
+    data.vote_selectors[category] = selector
+    return selector
+end)
+:on_selection_changed(function(player, element)
+    local data = player_data[player.index]
+
+    -- Find the category for this vote selector
+    local category
+    for selector_category, selector in pairs(data.vote_selectors) do
+        if selector == element then
+            category = selector_category
+        end
+    end
+    if not category then return end
```

Comment:
As there will only be one selector per category, I suggest setting the name of the wrapper flow to the category name.
This would simplify this to `local category = element.parent.name`